esphome/components/xgzp68xx/sensor.py

```diff
@@ -0,0 +1,65 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import i2c, sensor
+from esphome.const import (
+    DEVICE_CLASS_PRESSURE,
+    CONF_ID,
+    DEVICE_CLASS_TEMPERATURE,
+    STATE_CLASS_MEASUREMENT,
+    UNIT_PASCAL,
+    UNIT_CELSIUS,
+    CONF_TEMPERATURE,
+    CONF_PRESSURE,
+    CONF_OFFSET,
+)
+
+DEPENDENCIES = ["i2c"]
+CODEOWNERS = ["@gcormier"]
+
+CONF_K_VALUE = "k_value"
+
+xgzp68xx_ns = cg.esphome_ns.namespace("xgzp68xx")
+XGZP68XXComponent = xgzp68xx_ns.class_(
+    "XGZP68XXComponent", cg.PollingComponent, i2c.I2CDevice
+)
+
+CONFIG_SCHEMA = (
+    cv.Schema(
+        {
+            cv.GenerateID(): cv.declare_id(XGZP68XXComponent),
+            cv.Optional(CONF_PRESSURE): sensor.sensor_schema(
+                unit_of_measurement=UNIT_PASCAL,
+                accuracy_decimals=1,
+                device_class=DEVICE_CLASS_PRESSURE,
+                state_class=STATE_CLASS_MEASUREMENT,
+            ),
+            cv.Optional(CONF_TEMPERATURE): sensor.sensor_schema(
+                unit_of_measurement=UNIT_CELSIUS,
+                accuracy_decimals=1,
+                device_class=DEVICE_CLASS_TEMPERATURE,
+                state_class=STATE_CLASS_MEASUREMENT,
+            ),
+            cv.Optional(CONF_K_VALUE): cv.uint16_t,
```

Comment:
This should be a config default, not hidden inside `to_code`

```suggestion
            cv.Optional(CONF_K_VALUE, default=4096): cv.uint16_t,
```